File: app/src/main/java/com/fanwe/www/http/MainActivity.java
package com.fanwe.www.http;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import com.fanwe.lib.http.RequestManager;
import com.fanwe.lib.http.cookie.SerializableCookieStore;
import com.fanwe.lib.http.utils.LogUtil;

public class MainActivity extends AppCompatActivity
{
    public static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        LogUtil.setDebug(true);

        //设置cookie管理对象
        RequestManager.getInstance().setCookieStore(new SerializableCookieStore(this));

        //设置请求拦截对象，可用于log输出，或者一些需要全局处理的逻辑，注意这边传入的对象如果是和资源相关的对象，需要在资源销毁的时候remove
        RequestManager.getInstance().addRequestInterceptor(new AppRequestInterceptor());

        //设置Request的唯一标识生成对象，注意这边传入的对象不应该是和资源相关的对象，否则资源销毁回调的时候被单例持有会造成内存泄漏
        RequestManager.getInstance().setRequestIdentifierProvider(new AppRequestIdentifierProvider());
    }

    public void onClickAsyncRequestActivity(View view)
    {
        startActivity(new Intent(this, AsyncRequestActivity.class));
    }

    public void onClickSyncRequestActivity(View view)
    {
        startActivity(new Intent(this, SyncRequestActivity.class));
    }

    public void onClickDownloadActivity(View view)
    {
        startActivity(new Intent(this, DownloadActivity.class));
    }

    @Override
    protected void onDestroy()
    {
        super.onDestroy();
    }
}

File: app/src/main/java/com/sd/demo/http/activity/MainActivity.kt
package com.sd.demo.http.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.sd.lib.http.RequestManager
import com.sd.lib.http.cookie.store.SerializableCookieStore
import com.sd.demo.http.databinding.ActivityMainBinding
import com.sd.demo.http.utils.AppRequestInterceptor
import com.sd.demo.http.utils.AppResponseParser
import com.sd.demo.http.utils.AppResultInterceptor

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var _binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(_binding.root)

        // 调试模式，输出log
        RequestManager.isDebug = true

        // cookie管理对象
        RequestManager.cookieStore = SerializableCookieStore(this)

        // 请求拦截
        RequestManager.requestInterceptor = AppRequestInterceptor()

        // http返回解析器
        RequestManager.responseParser = AppResponseParser()

        // 结果拦截器
        RequestManager.resultInterceptor = AppResultInterceptor()
    }

    override fun onClick(v: View) {
        when (v) {
            _binding.btnAsyncRequestActivity -> startActivity(Intent(this, AsyncRequestActivity::class.java))
            _binding.btnSyncRequestActivity -> startActivity(Intent(this, SyncRequestActivity::class.java))
            _binding.btnDownloadActivity -> startActivity(Intent(this, DownloadActivity::class.java))
        }
    }
}